packages/components/badge/src/Badge.vue

```diff
@@ -2,58 +2,67 @@
   <span class="ix-badge-wrapper">
     <slot />
     <sup class="ix-badge" :class="classes" :style="styles">
-      <slot v-if="$slots.custom" name="custom" />
+      <slot v-if="slotsExist.count" name="count" />
       <template v-else-if="countValue">{{ countValue }}</template>
     </sup>
   </span>
 </template>
 <script lang="ts">
-import { computed, defineComponent, Slots } from 'vue'
-import { BadgeProps } from './types'
-import { isNumeric, isDefined } from './utils'
+import { computed, defineComponent, onUpdated, reactive } from 'vue'
+import { BadgeProps, SlotsExist } from './types'
+import { isNumeric } from './utils'
 
 export default defineComponent({
   name: 'IxBadge',
   props: {
-    dot: { type: Boolean, default: false },
     count: { type: [Number, String], default: 0 },
-    overflowCount: { type: [Number, String], default: 99 },
     showZero: { type: Boolean, default: false },
+    overflowCount: { type: [Number, String], default: 99 },
+    dot: { type: Boolean, default: false },
     color: { type: String, default: undefined },
   },
   setup(props: BadgeProps, { slots }) {
-    const countValue = useCountValue(props, slots)
+    const slotsExist = reactive({
+      default: !!slots.default,
+      count: !!slots.count,
+    })
+    onUpdated(() => {
+      slotsExist.default = !!slots.default
+      slotsExist.count = !!slots.count
+    })
+
+    const countValue = useCountValue(props, slotsExist)
 
-    const classes = useClasses(props, slots)
+    const classes = useClasses(props, slotsExist)
 
-    const styles = useStyles(props, slots)
+    const styles = useStyles(props, slotsExist)
 
     return {
       countValue,
       classes,
       styles,
+      slotsExist,
     }
   },
 })
 
-const useCountValue = (props: BadgeProps, slots: Slots) => {
+const useCountValue = (props: BadgeProps, slots: SlotsExist) => {
   return computed(() => {
-    if (!slots.custom && !props.dot) {
+    if (!slots.count && !props.dot) {
       if (!props.showZero && +props.count === 0) return false
       if (isNumeric(props.count) && isNumeric(props.overflowCount)) {
-        return +props.count > +props.overflowCount ? `${props.overflowCount}+` : `${props.count}`
+        return props.count > props.overflowCount ? `${props.overflowCount}+` : `${props.count}`
       }
       return props.count
     }
     return false
   })
 }
 
-const useStyles = (props: BadgeProps, slots: Slots) => {
+const useStyles = (props: BadgeProps, slots: SlotsExist) => {
   return computed(() => {
-    const { custom: customSlot } = slots
-    const color = isDefined(props.color) && props.color
-    if (customSlot) {
+    const color = props?.color ?? ''
```

Comment:
```suggestion
    const color = props.color ?? ''
```
这里的 ？ 好像没意义， props 不可能为 undefined